Router.svelte

```diff
@@ -157,15 +157,36 @@ export function link(node) {
     // Destination must start with '/'
     const href = node.getAttribute('href')
     if (!href || href.length < 1 || href.charAt(0) != '/') {
-        throw Error('Invalid value for "href" attribute')
+        throw Error('Invalid value for "href" attribute: ' + href)
     }
 
     // Add # to every href attribute
     node.setAttribute('href', '#' + href)
+
+    // If a user wants to restore scroll positions, we do this to know when they are about to navigate to a new 'page'
+    // TODO: it seems I don't have reference to this var from context='module'; will need to fix this
+    //if (restoreScrollPosition) {
+    node.addEventListener('click', scrollstateHistoryHandler)
+    //}
+}
+
+/**
+ * Handler attached to an anchor tag (above)
+ *
+ * @param {HTMLElementEventMap} event - an onclick event attached to an anchor tag
+ */
+function scrollstateHistoryHandler(event) {
```

Comment:
Please move this to be a lambda function for the `node.addEventListener` invocation